crates/curp/src/server/storage/wal/pipeline.rs

```diff
@@ -0,0 +1,183 @@
+use std::{
+    io,
+    path::{Path, PathBuf},
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc,
+    },
+    task::Poll,
+};
+
+use clippy_utilities::OverflowArithmetic;
+use event_listener::Event;
+use flume::r#async::RecvStream;
+use futures::{FutureExt, StreamExt};
+use thiserror::Error;
+use tokio::task::JoinHandle;
+use tokio_stream::Stream;
+use tracing::error;
+
+use super::util::LockedFile;
+
+/// The temp file extension
+const TEMP_FILE_EXT: &str = ".tmp";
+
+/// The file pipeline, used for pipelining the creation of temp file
+pub(super) struct FilePipeline {
```

Comment:
Can a user create more than one `FilePipeline`? If so, what will happen?

Reply:
The user don't create a pipeline, it only act as a WAL componet and will only be created once.